### PR TITLE
test: 1452 refactor modal pages batch 2

### PR DESCRIPTION
**Pull request by SamuelSalas**
## **Description**
We are aiming to refactor the page objects in the modal folder so that they strictly follow the page object model pattern. This would aide in providing more readable and help standardize the way we create our tests. Because of the amount of files inside of the modal folder, the scope was divided into a batch of 4.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: e2e/specs/quarantine/deeplinks.failing.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ import Accounts from '../../../wdio/helpers/Accounts';
 import TabBarComponent from '../../pages/TabBarComponent';
 import Assertions from '../../utils/Assertions';
 
-const BINANCE_RPC_URL = 'https://bsc-dataseed1.binance.org';
+//const BINANCE_RPC_URL = 'https://bsc-dataseed1.binance.org';
```
Comment (cortisiko) [inline e2e/specs/quarantine/deeplinks.failing.js:23]: remove comment

File: e2e/specs/quarantine/deeplinks.failing.js [hunk 2]
```diff
@@ -98,13 +98,16 @@ describe(Regression('Deep linking Tests'), () => {
...
-    await NetworkApprovalModal.isVisible();
-    await NetworkApprovalModal.isDisplayNameVisible('BNB Smart Chain');
-    await NetworkApprovalModal.isNetworkURLVisible(BINANCE_RPC_URL);
-    await NetworkApprovalModal.isChainIDVisible('56');
+    await Assertions.checkIfVisible(NetworkApprovalModal.container);
+    await Assertions.checkIfElementToHaveText(
+      NetworkApprovalModal.displayName,
+      'BNB Smart Chain',
+    );
+    //await NetworkApprovalModal.isNetworkURLVisible(BINANCE_RPC_URL);
```
Comment (cortisiko) [inline e2e/specs/quarantine/deeplinks.failing.js:106]: remove comment

File: e2e/specs/quarantine/deeplinks.failing.js [hunk 3]
```diff
@@ -114,14 +117,17 @@ describe(Regression('Deep linking Tests'), () => {
...
 
-    await NetworkApprovalModal.isVisible();
-    await NetworkApprovalModal.isDisplayNameVisible('Polygon Mainnet');
-    await NetworkApprovalModal.isChainIDVisible('137');
+    await Assertions.checkIfVisible(NetworkApprovalModal.container);
+    await Assertions.checkIfElementToHaveText(
+      NetworkApprovalModal.displayName,
+      'Polygon Mainnet',
+    );
+    //await NetworkApprovalModal.isChainIDVisible('137');
```
Comment (cortisiko) [inline e2e/specs/quarantine/deeplinks.failing.js:125]: remove comment

File: e2e/pages/modals/NetworkApprovalModal.js [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
-  static async isChainIDVisible(chainID) {
-    await TestHelpers.checkIfElementWithTextIsVisible(chainID);
+  async tapApproveButton() {
+    await Gestures.waitAndTap(this.approvedButton);
   }
 
-  static async isNetworkURLVisible(networkURL) {
-    await TestHelpers.checkIfElementWithTextIsVisible(networkURL);
-  }
+  //static async isChainIDVisible(chainID) {
```
Comment (cortisiko) [inline e2e/pages/modals/NetworkApprovalModal.js:26]: remove comments: line 26-32

**Review by cortisiko — CHANGES_REQUESTED** [review]
left some comments

**Comment by SamuelSalas** [comment]
Regression Test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/701c4020-c70a-466c-8606-3c1066259c29
Smoke Test run: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/4e18f36a-cf2e-4eeb-bd1b-82b5ee1e55fc
Appium wdio test run: https://app.bitrise.io/build/9d369fcc-d944-48c2-a00f-0352d4961708

**Review by cortisiko — APPROVED** [review]
✅ ✅ ✅ ✅

**Comment by chrisleewilcox** [comment]
Regression: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/3bedadae-3558-48ac-9f61-197febd26064

**Review by Andepande — APPROVED** [review]
LGTM

**Pull request merged**